src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import Loading from '../../Loading/Loading';
+
+const ReportsTrafficSources = function({
+  domainName,
+  dateFrom,
+  intl,
+  dependencies: { datahubClient },
+}) {
+  const [trafficSources, setTrafficSources] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const trafficSourcesData = await datahubClient.getTrafficSourcesByPeriod(
+        domainName,
+        dateFrom,
+      );
+      setTrafficSources(trafficSourcesData);
+      let total = 0;
+      trafficSources.forEach((item, index) => {
+        total += item.quantity;
+      });
```

Comment:
Please, use _reduce_ in place of an imperative iteration:

```js
const total = trafficSources.reduce((accumulator, item) => accumulator + item.quatity);
```

Reply:
i changed